### PR TITLE
bug(BILL-CPC-920) Implement Negative Testing In BillServiceClientIntegration tests

### PR DESCRIPTION
**Pull request by drallo22**
JIRA: (https://champlainsaintlambert.atlassian.net/jira/software/c/projects/CPC/boards/35?selectedIssue=CPC-920&sprints=315)
## Context:
This ticket involves adding negative tests to the BillServiceClient integration tests to assess how the system behaves under the wrong conditions, ensuring reliability and error handling.
###  Created 7 negative tests

- getNonExistentBillById
- getBillByInvalidVetId
- getBillsByInvalidCustomerId
- deleteNonExistentBill
- deleteBillWithInvalidVetId
- deleteBillWithInvalidCustomerId
- createBillWithInvalidRequest

### CODE REVIEW

**Review by dahlica12 — APPROVED** [review]
Good use of testing which will provide more coverage and make the environment more robust. Correct naming conventions used and overall good purpose!

File: api-gateway/src/test/java/com/petclinic/bffapigateway/domainclientlayer/BillServiceClientIntegrationTest.java [hunk 1]
```diff
@@ -369,5 +370,89 @@ void shouldUpdateSpecificFieldsOfBill() throws Exception {
...
+    @Test
+    void getBillsByInvalidCustomerId() {
+        server.enqueue(new MockResponse().setResponseCode(500));
+
+        Flux<BillResponseDTO> billResponseDTOMono = billServiceClient.getBillsByOwnerId("invalidCustomerId");
+
+        StepVerifier.create(billResponseDTOMono)
+                .expectError(WebClientResponseException.InternalServerError.class)
+                .verify();
+    }
```
Comment (ebond07) [inline api-gateway/src/test/java/com/petclinic/bffapigateway/domainclientlayer/BillServiceClientIntegrationTest.java:404]: These two tests seem extremely similar. Should they be throwing different response codes?

Comment (drallo22) [reply api-gateway/src/test/java/com/petclinic/bffapigateway/domainclientlayer/BillServiceClientIntegrationTest.java]: Yes, thank  you for catching that I have updated it to excpect 400 as it should.

**Review by ebond07 — APPROVED** [review]

**Comment by ebond07** [comment]
Good job, thanks for considering my comment.

**Review by Youssefchahboune — APPROVED** [review]

**Pull request merged**